src/attributes/classes.js

```diff
@@ -114,22 +114,31 @@ jQuery.fn.extend({
 		}
 
 		return this.each(function() {
-			var className, i, self, classNames;
+			var className, i, self, classNames, predicate;
 
 			if ( type === "string" ) {
 
 				// Toggle individual class names
 				i = 0;
 				self = jQuery( this );
+				predicate = jQuery.isFunction( stateVal ) ? stateVal.call( self, i, self) : null;
```

Comment:
@dmethvin Good points. @jdphenix, would you want to try to document this thing? The docs live in https://github.com/jquery/api.jquery.com/